client/cli/src/lib.rs

```diff
@@ -315,19 +302,61 @@ pub fn init_logger(
 		"%Y-%m-%d %H:%M:%S%.3f".to_string()
 	});
 
-	let subscriber = FmtSubscriber::builder()
+	let subscriber_builder = FmtSubscriber::builder()
 		.with_env_filter(env_filter)
-		.with_writer(std::io::stderr)
+		.with_writer(std::io::stderr as _)
 		.event_format(logging::EventFormat {
 			timer,
+			enable_color,
 			display_target: !simple,
 			display_level: !simple,
 			display_thread_name: !simple,
-			enable_color,
-		})
-		.finish()
-		.with(logging::NodeNameLayer);
+		});
+	if disable_log_reloading {
+		let subscriber = subscriber_builder
+			.finish()
+			.with(logging::NodeNameLayer);
+		initialize_tracing(subscriber, tracing_receiver, profiling_targets)
+	} else {
+		let subscriber_builder = subscriber_builder.with_filter_reloading();
+		let handle = subscriber_builder.reload_handle();
+		sc_tracing::set_reload_handle(handle);
+		let subscriber = subscriber_builder
+			.finish()
+			.with(logging::NodeNameLayer);
+		initialize_tracing(subscriber, tracing_receiver, profiling_targets)
+	}
+}
+
+// Adds default directives to ensure setLogFilter RPC functions correctly
+fn add_default_directives(mut env_filter: EnvFilter, directives: &str) -> EnvFilter {
+	use sc_tracing::{parse_directives, add_default_directives};
+	add_default_directives(directives);
+	let (oks, errs): (Vec<_>, Vec<_>) = parse_directives(directives)
+		.into_iter()
+		.partition(|res| res.is_ok());
+	for invalid_directive in errs {
```

Comment:
Before it panicked if the default ones are invalid and we should keep this behavior.